gopackagesdriver fails to build

`gopackagesdriver` is used to allow the language server (gopls) to resolve go libraries managed by bazel. The setup instructions is documented in https://github.com/bazelbuild/rules_go/wiki/Editor-setup, however following it results in the standard library packages unable to be resolved.

Running the below command gives more information
```
koonpeng@koonpeng:~/ws/googlecloudrobotics/core$ echo {} | bazel run -- @io_bazel_rules_go//go/tools/gopackagesdriver 
std
...
ERROR: /usr/local/google/home/koonpeng/.cache/bazel/_bazel_koonpeng/ae7eebad1ef4062c3ff22e4722fa5b41/external/io_bazel_rules_go/BUILD.bazel:42:7: GoStdlibList external/io_bazel_rules_go/stdlib_/stdlib.pkg.json failed: (Exit 1): builder failed: error executing command (from target @io_bazel_rules_go//:stdlib) bazel-out/k8-opt-exec-2B5CBBC6/bin/external/go_sdk/builder_reset/builder stdliblist -sdk external/go_sdk -installsuffix linux_amd64 -sdk external/go_sdk -out ... (remaining 3 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
# runtime/cgo
_cgo_export.c:3:10: fatal error: 'stdlib.h' file not found
stdliblist: error running subcommand external/go_sdk/bin/go: exit status 1
{"NotHandled":false,"Compiler":"gc","Arch":"amd64","Packages":[]}
```

`gopackagesdriver` is trying to build *something* which fails with `stdlib.h not found`. Our custom toolchain might be causing the issue, according to https://github.com/bazelbuild/rules_go/issues/3886#issuecomment-2010865086, the CGO flags should be converted to absolute paths but for some reason it is not happening.

```
koonpeng@koonpeng:~/ws/googlecloudrobotics/core$ bazel build --subcommands --aspects=@io_bazel_rules_go//go/tools/go
packagesdriver:aspect.bzl%go_pkg_info_aspect --output_groups=go_pkg_driver_json_file,go_pkg_driver_stdlib_json_file,
go_pkg_driver_srcs @io_bazel_rules_go//:stdlib
...
(cd /usr/local/google/home/koonpeng/.cache/bazel/_bazel_koonpeng/ae7eebad1ef4062c3ff22e4722fa5b41/execroot/cloud_robotics && \
  exec env - \
    CC=external/llvm_toolchain/bin/cc_wrapper.sh \
    CGO_CFLAGS='-U_FORTIFY_SOURCE --target=x86_64-unknown-linux-gnu -U_FORTIFY_SOURCE -fstack-protector -fno-omit-frame-pointer -Wthread-safety -Wself-assign --sysroot=external/com_googleapis_storage_chrome_linux_amd64_sysroot -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" -fdebug-prefix-map=external/llvm_toolchain_llvm/=__bazel_toolchain_llvm_repo__/' \
    CGO_ENABLED=1 \
    CGO_LDFLAGS='--target=x86_64-unknown-linux-gnu -lm -no-canonical-prefixes -fuse-ld=lld -Wl,--build-id=md5 -Wl,--hash-style=gnu -Wl,-z,relro,-z,now -l:libc++.a -l:libc++abi.a -l:libunwind.a -rtlib=compiler-rt -lpthread -ldl --sysroot=external/com_googleapis_storage_chrome_linux_amd64_sysroot' \
    GOARCH=amd64 \
    GOEXPERIMENT=nocoverageredesign \
    GOOS=linux \
    GOPATH='' \
    GOROOT=external/go_sdk \
    GOROOT_FINAL=GOROOT \
    GOTOOLCHAIN=local \
    PATH=external/llvm_toolchain/bin:/bin:/usr/bin \
  bazel-out/k8-opt-exec-2B5CBBC6/bin/external/go_sdk/builder_reset/builder stdliblist -sdk external/go_sdk -installsuffix linux_amd64 -sdk external/go_sdk -out bazel-out/k8-fastbuild/bin/external/io_bazel_rules_go/stdlib_/stdlib.pkg.json -cache bazel-out/k8-fastbuild/bin/external/io_bazel_rules_go/stdlib_/gocache)
```

(notice that the sysroot is using relative path)

Disabling `--incompatible_enable_cc_toolchain_resolution` and `--action_env=BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1` fixes the issue as it removes the `--sysroot` flag, so for now we can use this to workaround.

```
GOPACKAGESDRIVER_BAZEL_BUILD_FLAGS="--noincompatible_enable_cc_toolchain_resolution --action_env=BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=0" exec bazel run -- @io_bazel_rules_go//go/tools/gopackagesdriver "${@}"
```

--- a/bazel/repositories.bzl
+++ b/bazel/repositories.bzl
@@ -49,10 +49,10 @@ def cloud_robotics_repositories():
     _maybe(
         http_archive,
         name = "io_bazel_rules_go",
-        sha256 = "91585017debb61982f7054c9688857a2ad1fd823fc3f9cb05048b0025c47d023",
+        sha256 = "80a98277ad1311dacd837f9b16db62887702e9f1d1c4c9f796d0121a46c8e184",
         urls = [
-            "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.42.0/rules_go-v0.42.0.zip",
-            "https://github.com/bazelbuild/rules_go/releases/download/v0.42.0/rules_go-v0.42.0.zip",
+            "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.46.0/rules_go-v0.46.0.zip",
+            "https://github.com/bazelbuild/rules_go/releases/download/v0.46.0/rules_go-v0.46.0.zip",
         ],
     )
     _maybe(

